File: Third_Year/DS/Practicas/P2/codigo_flutter/lib/GestorFiltros.dart
import 'FiltroIVA.dart';
import 'FiltroPersonas.dart';
import 'FiltroTiempo.dart';
import 'Alquiler.dart';
import 'CadenaFiltro.dart';
import 'Filtro.dart';

class GestorFiltros {
  CadenaFiltro _cadenaFiltros = new CadenaFiltro();

  GestorFiltros() {
    FiltroTiempo f1 = new FiltroTiempo();
    FiltroPersonas f2 = new FiltroPersonas();
    FiltroIVA f3 = new FiltroIVA();

    _cadenaFiltros.addFiltro(f1);
    _cadenaFiltros.addFiltro(f2);
    _cadenaFiltros.addFiltro(f3);
  }

  void filtrar(Alquiler alq) {
    _cadenaFiltros.ejecutar(alq);
  }

  void setFiltro(Filtro f) {
    _cadenaFiltros.addFiltro(f);
  }
}

File: Third_Year/DS/Practicas/P1/Filtros/GestorFiltros.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

#include "GestorFiltros.h"

GestorFiltros::GestorFiltros(){
    FiltroTiempo * f1 = new FiltroTiempo();
    FiltroPersonas * f2 = new FiltroPersonas();
    FiltroIva * f3 = new FiltroIva();

    cadenaFiltros.addFiltro(f1);
    cadenaFiltros.addFiltro(f2);
    cadenaFiltros.addFiltro(f3);
}

void GestorFiltros::filtrar(Alquiler & alq){
    cadenaFiltros.ejecutar(alq);
}

void GestorFiltros::setFiltro(Filtro * f){
    cadenaFiltros.addFiltro(f);
}